x-pack/plugins/infra/public/hooks/use_breadcrumbs.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ChromeBreadcrumb } from 'kibana/public';
+import { i18n } from '@kbn/i18n';
+import { MouseEvent, useEffect } from 'react';
+import { EuiBreadcrumb } from '@elastic/eui';
+import { useKibana } from '../../../../../src/plugins/kibana_react/public';
+
+function handleBreadcrumbClick(
```

Comment:
I plan to remove this and use `useLinkProps` instead as suggested by @weltenwort

Reply:
👌  Yeah, `useLinkProps` is the way to go.

Reply:
@weltenwort Yep makes sense. I made use of `useLinkProps` on a higher level here https://github.com/elastic/kibana/pull/103249/commits/5d9ce0b99e4c734fa9ebe3ea4d87928926971b7b. Metrics detail page is the only case that needs the useLinkProps, since rest pages are using the base app-level breadcrumbs (`Observability > Logs` and `Observability > Metrics`) and last breadcrumb is not clickable anyway for the subpages (Inventory, Streams etc).

Thus being said, I need to keep above function for the app-level breadcrumbs, but I could do a slight modification. Instead of

```
setBreadcrumbs(
        handleBreadcrumbClick(
          makeBaseBreadcrumb(path, observabilityPath, appTitle).concat(extraCrumbs),
          navigate
        )
      );
```

I could use:

```
setBreadcrumbs(
          handleBreadcrumbClick(makeBaseBreadcrumb(path, observabilityPath, appTitle)).concat(extraCrumbs),
          navigate
        )
      );
```

What do you think? My reasoning about keeping it was that a higher level component could theoretically pass an `href` attribute to any object when calling useBreadCrumb and the element would automatically become clickable. But I agree it is better to be consistent and use `useLinkProps` on a higher level, which will take care of adding the link plus more stuff like params, hash etc. Shall I do the change I suggested above? Let me know if it's not clear what I am talking about.

Reply:
It's not obvious to me why `useLinkProps` in `useBreadcrumbs` wouldn't work. My intuition tells me it would look something like the following:

```typescript
export const useBreadcrumbs = (app: AppId, appTitle: string, extraCrumbs: ChromeBreadcrumb[]) => {
  const {
    services: { chrome },
  } = useKibanaContextForPlugin();

  const observabilityLinkProps = useLinkProps({ app: 'observability-overview' });
  const appLinkProps = useLinkProps({ app });

  useEffect(() => {
    chrome?.setBreadcrumbs?.([
      {
        ...observabilityLinkProps,
        text: observabilityTitle,
      },
      {
        ...appLinkProps,
        text: appTitle,
      },
      ...extraCrumbs,
    ]);
  }, [appLinkProps, appTitle, chrome, extraCrumbs, observabilityLinkProps]);
};

```

It's untested, though, so maybe I've overlooked something? :shrug:

Reply:
I tried out this code, works fine!